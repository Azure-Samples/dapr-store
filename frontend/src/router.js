import VueRouter from 'vue-router'

import ProductCatalog from './components/ProductCatalog'
import ProductOffers from './components/ProductOffers'
import Home from './components/Home'

const router = new VueRouter({
  routes: [
    {
      path: '/catalog',
      component: ProductCatalog
    },
    {
      path: '/offers',
      component: ProductOffers
    },
    {
      path: '/',
      component: Home
    }
  ]
})

export default router