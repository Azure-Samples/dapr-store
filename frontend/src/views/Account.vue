<template>
  <div>
    <h1>
      User Account
      <b-button size="lg" variant="danger" class="float-right" @click="logout">
        <fa icon="sign-out-alt" /> &nbsp; LOGOUT
      </b-button>
    </h1>

    <table v-if="user.account && user.userName" class="table">
      <tr><td>Display Name</td><td>{{ user.account.name }}</td></tr>
      <tr><td>Username</td><td>{{ user.userName }}</td></tr>
      <tr><td>Unique ID</td><td>{{ user.account.accountIdentifier }}</td></tr>
    </table>

    <div />
  </div>
</template>

<script>
import { userProfile } from '../main'
import User from '../user'

export default {
  name: 'Account',

  data() {
    return {
      user: userProfile
    }
  },

  created() {
    if (!userProfile.token) {
      this.$router.replace({ path: '/' })
    }
  },

  methods: {
    async logout() {
      Object.assign(userProfile, new User())
      localStorage.removeItem('user')
      localStorage.removeItem('cart')
      //await msalApp.logout()

      this.$router.push({ name: 'home' })
    }
  }
}
</script>

<style scoped>
.table {
  font-size: 140%;
}
</style>