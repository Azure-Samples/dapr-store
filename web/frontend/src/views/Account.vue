<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - User / account details view
// ----------------------------------------------------------------------------
-->

<template>
  <div>
    <error-box :error="error" />
    <h1>
      User Account
      <b-button size="lg" variant="danger" class="float-right" @click="logout">
        <fa icon="sign-out-alt" /> &nbsp; LOGOUT
      </b-button>
    </h1>
    <br>

    <div v-if="!registeredUser" class="text-center">
      <b-spinner variant="success" style="width: 5rem; height: 5rem;" />
    </div>
    <b-card v-if="registeredUser" class="details">
      <img class="profile d-none d-md-block" :src="photo">

      Display Name: <b>{{ registeredUser.displayName }}</b>
      <br>
      Username: <b>{{ registeredUser.username }}</b>
    </b-card>

    <br>
    <h1>
      Orders
      <b-button size="lg" variant="success" class="float-right" @click="reloadOrders">
        <fa icon="redo-alt" /> &nbsp; Refresh
      </b-button>
    </h1>

    <div v-if="!ordersLoaded" class="text-center">
      <b-spinner variant="success" style="width: 5rem; height: 5rem;" />
    </div>

    <order v-for="order in orders" :key="order.id" :order="order" />
  </div>
</template>

<script>
import api from '../services/api'
import auth from '../services/auth'
import graph from '../services/graph'
import ErrorBox from '../components/ErrorBox'
import Order from '../components/Order'

export default {
  name: 'Account',

  components: {
    'error-box': ErrorBox,
    'order': Order
  },

  data() {
    return {
      registeredUser: auth.user(),
      photo: 'img/placeholder-profile.jpg',
      error: null,
      orders: null,
      ordersLoaded: false
    }
  },

  async created() {
    try {
      if (auth.user()) {
        let resp = await api.userGet(auth.user().userName)
        if (resp) {
          this.registeredUser = resp
        }
        graph.getPhoto()
          .then((photo) => { if (photo){ this.photo = photo } })
      }
    } catch (err) {
      this.error = err
    }

    this.reloadOrders()
  },

  methods: {
    async logout() {
      await auth.logout()

      this.$router.push({ name: 'home' })
    },

    async reloadOrders() {
      this.ordersLoaded = false
      this.orders = []
      let orderList = []

      try {
        orderList = await api.ordersForUser(auth.user().userName)
      } catch (err) {
        this.error = err
      }

      // If you have no orders, skip it
      if (!orderList) {
        this.ordersLoaded = true
        return
      }

      // Load orders call the API to fetch details
      for (let orderId of orderList.reverse()) {
        try {
          let order = await api.orderGet(orderId)
          this.orders.push(order)
        } catch (err) {
          this.error += err
          continue
        }
      }

      this.ordersLoaded = true
    }
  }
}
</script>

<style scoped>
  code {
    color:rgb(23, 38, 173);
    font-size: 1.2rem;
  }

  .details {
    font-size: 140%;
  }

  .profile {
    float: right;
    width: 10rem;
    border-radius: 50%;
  }
</style>