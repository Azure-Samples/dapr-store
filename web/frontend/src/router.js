// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - client side routes
// ----------------------------------------------------------------------------

import { createRouter, createWebHistory } from 'vue-router'

import auth from './services/auth'

import ProductCatalog from './views/ProductCatalog.vue'
import ProductOffers from './views/ProductOffers.vue'
import ProductSearch from './views/ProductSearch.vue'
import ProductSingle from './views/ProductSingle.vue'
import ViewOrder from './views/ViewOrder.vue'
import Login from './views/Login.vue'
import Account from './views/Account.vue'
import Home from './views/Home.vue'
import About from './views/About.vue'
import Cart from './views/Cart.vue'

const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      name: 'home',
      path: '/',
      component: Home
    },
    {
      name: 'login',
      path: '/login',
      component: Login
    },
    {
      name: 'account',
      path: '/account',
      component: Account,
      beforeEnter: signedInCheck
    },
    {
      name: 'cart',
      path: '/cart',
      component: Cart,
      beforeEnter: signedInCheck
    },
    {
      name: 'about',
      path: '/about',
      component: About
    },
    {
      name: 'search',
      path: '/search/:query',
      component: ProductSearch
    },
    {
      name: 'catalog',
      path: '/catalog',
      component: ProductCatalog
    },
    {
      name: 'offers',
      path: '/offers',
      component: ProductOffers
    },
    {
      name: 'single-product',
      path: '/product/:id',
      component: ProductSingle
    },
    {
      name: 'view-order',
      path: '/order/:id',
      component: ViewOrder
      //beforeEnter: signedInCheck
    }
  ]
})

function signedInCheck(to, from, next) {
  const user = auth.user()
  if (!user || !user.username) {
    next('/login')
  } else {
    next()
  }
}

export default router
