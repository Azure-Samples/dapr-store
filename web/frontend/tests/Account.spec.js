import { describe, it, vi, expect } from 'vitest'
import { flushPromises, mount } from '@vue/test-utils'
import router from '@/router'

import Account from '@/views/Account.vue'

vi.mock('@/services/auth')
vi.mock('@/services/api')

describe('Account.vue', () => {
  it('renders user profile', async () => {
    const wrapper = mount(Account, {
      global: {
        plugins: [router]
      }
    })

    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
