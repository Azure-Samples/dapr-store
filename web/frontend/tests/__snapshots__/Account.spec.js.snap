// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Account.vue > renders user profile 1`] = `
"<div data-v-48d1de81="">
  <!--v-if-->
  <div data-v-48d1de81="" class="d-flex justify-content-between mb-2">
    <h1 data-v-48d1de81="">User Account</h1><button data-v-48d1de81="" class="btn btn-danger" style="height: 3rem;"><i data-v-48d1de81="" class="fa-solid fa-right-from-bracket"></i> &nbsp; LOGOUT</button>
  </div>
  <!--v-if-->
  <div data-v-48d1de81="" class="card mb-5">
    <div data-v-48d1de81="" class="card-body">
      <div data-v-48d1de81="" class="d-flex justify-content-between mb-2">
        <table data-v-48d1de81="" class="details">
          <tr data-v-48d1de81="">
            <td data-v-48d1de81="">Display Name</td>
            <td data-v-48d1de81="">Mock User</td>
          </tr>
          <tr data-v-48d1de81="">
            <td data-v-48d1de81="">Username</td>
            <td data-v-48d1de81="">demo@example.net</td>
          </tr>
          <tr data-v-48d1de81="">
            <td data-v-48d1de81="">Environment</td>
            <td data-v-48d1de81="">Fake</td>
          </tr>
        </table><img data-v-48d1de81="" class="profile d-md-block" src="img/placeholder-profile.jpg">
      </div>
    </div>
  </div>
  <div data-v-48d1de81="" class="d-flex justify-content-between">
    <h1 data-v-48d1de81="">Orders</h1>
    <div data-v-48d1de81="" class="btn btn-success" style="height: 3rem;"><i data-v-48d1de81="" class="fa-solid fa-rotate"></i> &nbsp; Refresh</div>
  </div>
  <!--v-if-->
  <div data-v-f415f84a="" data-v-48d1de81="" class="card order p-4">
    <div data-v-f415f84a="" class="d-flex justify-content-between mb-2">
      <h2 data-v-f415f84a="">Mock order</h2><a data-v-f415f84a="" href="/order/ord-mock" class="btn btn-primary"><i data-v-f415f84a="" class="fa-solid fa-list-check"></i> &nbsp; ORDER DETAILS </a>
    </div>
    <h2 data-v-f415f84a=""> Status: <span data-v-f415f84a="" class="text-capitalize order-status order-new">new</span></h2>
    <ul data-v-f415f84a="">
      <li data-v-f415f84a="">Order Id: ord-mock</li>
      <li data-v-f415f84a="">Amount: £22.4</li>
      <li data-v-f415f84a="">Items:</li>
      <ul data-v-f415f84a="">
        <li data-v-f415f84a="">2 x Paisley Cravat Ascot Tie — £11.2</li>
      </ul>
    </ul>
  </div>
</div>"
`;
